modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/programs/Download.scala

```diff
@@ -1,30 +1,62 @@
 package org.tessellation.sdk.infrastructure.snapshot.programs
 
+import cats.Monad
 import cats.effect.Async
 import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import scala.concurrent.duration.{Duration, DurationInt}
 
 import org.tessellation.schema.node.NodeState
+import org.tessellation.schema.peer.Peer
 import org.tessellation.sdk.domain.node.NodeStorage
 import org.tessellation.sdk.infrastructure.snapshot.SnapshotConsensus
 
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
 object Download {
 
   def make[F[_]: Async](
     nodeStorage: NodeStorage[F],
-    consensus: SnapshotConsensus[F, _, _, _, _, _]
+    consensus: SnapshotConsensus[F, _, _, _, _, _],
+    getPeers: () => F[Set[Peer]]
   ): Download[F] =
     new Download[F](
       nodeStorage,
-      consensus
+      consensus,
+      getPeers
     ) {}
 }
 
 sealed abstract class Download[F[_]: Async] private (
   nodeStorage: NodeStorage[F],
-  consensus: SnapshotConsensus[F, _, _, _, _, _]
+  consensus: SnapshotConsensus[F, _, _, _, _, _],
+  getPeers: () => F[Set[Peer]]
```

Comment:
This notation makes no sense. `F` is already lazy and it's a blueprint, everytime you execute this `F` wrapped logic it's gonna execute from scratch. Adding `() => F[...]` doesn't make it re execute everytime it is invoked, `F` here has that property already. So I think you should have this defined as `getPeers: F[Set[Peer]]`.

Reply:
@TheMMaciek Yes that's correct. However, in another PR I was asked not to pass in `F[A]` so we're following the same pattern.